packages/syft/src/syft/service/worker/worker_pool_service.py

```diff
@@ -324,24 +325,41 @@ def _get_worker_container(
         )
 
 
-def _get_worker_container_status(
-    worker: SyftWorker, docker_client: Optional[docker.DockerClient] = None
-) -> Union[WorkerStatus, SyftError]:
-    container = _get_worker_container(worker, docker_client)
-    if isinstance(container, SyftError):
-        return container
-
-    container_status = container.status
-    syft_container_status = None
-    if container_status == "running":
-        syft_container_status = WorkerStatus.RUNNING
-    elif container_status in ["paused", "removing", "exited", "dead"]:
-        syft_container_status = WorkerStatus.STOPPED
-    elif container_status == "restarting":
-        syft_container_status = WorkerStatus.RESTARTED
-    elif container_status == "created":
-        syft_container_status = WorkerStatus.PENDING
-    else:
-        return SyftError(message=f"Unknown container status: {container_status}")
-
-    return syft_container_status
+_CONTAINER_STATUS_TO_WORKER_STATUS: Dict[str, WorkerStatus] = dict(
+    [
+        ("running", WorkerStatus.RUNNING),
+        *(
+            (status, WorkerStatus.STOPPED)
+            for status in ["paused", "removing", "exited", "dead"]
+        ),
+        ("restarting", WorkerStatus.RESTARTED),
+        ("created", WorkerStatus.PENDING),
+    ]
+)
+
+
+def _stop_worker_container(
+    worker: SyftWorker,
+    container: Container,
+    force: bool,
+) -> Optional[SyftError]:
+    try:
+        # stop the container
+        container.stop()
+        # Remove the container and its volumes
+        _remove_worker_container(container, force=force, v=True)
+    except Exception as e:
+        return SyftError(
+            message=f"Failed to delete worker with id: {worker.id}. Error: {e}"
+        )
+
+
+def _remove_worker_container(container: Container, **kwargs: Any) -> None:
+    try:
+        container.remove(**kwargs)
+    except docker.errors.NotFound:
+        return
+    except docker.errors.APIError as e:
+        if "removal of container" in str(e) and "is already in progress" in str(e):
+            # If the container is already being removed, ignore the error
+            return
```

Comment:
@rasswanth-s This is a more robust way to remove the container. Error message from docker server is dependent on the specific docker server implementation. I tried with another dockerd not Docker Desktop and this does not correctly detect the container no longer existing. Checking for a `docker.errors.NotFound` (which is the same as a 404 `docker.errors.APIError`) is more reliable.